Bazel build support

njlr opened the pull request.

njlr commented:
@googlebot I signed it!

ben-clayton commented on BUILD:
```diff
@@ -0,0 +1,35 @@
+cc_library(
```
Please add copyright headers to these new files

njlr replied:
Sure, can you point me to what I should copy-paste in?

ben-clayton replied:
See any other file, substituting the line comment characters if necessary 
Thanks!

pmuetschard commented on BUILD:
```diff
@@ -0,0 +1,35 @@
+    "src/**/*_test.cpp",
+  ])),
+  linkopts = select({
+    "@bazel_tools//src/conditions:linux_x86_64": [ "-pthread" ],
```
@ben-clayton what other platforms will need libpthread? I'm assuming at least the Android ones will need it as well.

ben-clayton replied:
Likely, yes, but I'm happy to evolve the build rules than try to get them all perfect first time.

ben-clayton commented:
Looks good, thank you.
The CLA bot still seems grumpy. Are sure the form submit succeeded?

Cheers,
Ben

njlr commented:
@googlebot I signed it!

njlr commented:
> Looks good, thank you.
> The CLA bot still seems grumpy. Are sure the form submit succeeded?
> 
> Cheers,
> Ben

It is signed according to https://cla.developers.google.com/clas

ben-clayton commented:
> It is signed according to https://cla.developers.google.com/clas

Odd. I've tried our internal checking tool and it doesn't seem to have an entry for your GitHub username. Is it possible that the email address you used to sign the CLA is different from the one associated with your GitHub account?

Cheers,
Ben

njlr commented:
> > It is signed according to https://cla.developers.google.com/clas
> 
> Odd. I've tried our internal checking tool and it doesn't seem to have an entry for your GitHub username. Is it possible that the email address you used to sign the CLA is different from the one associated with your GitHub account?
> 
> Cheers,
> Ben

I changed my username at one point. I have updated the form with the new one to be safe. Perhaps it works now?

ben-clayton commented:
It might need to be tickled with the:

@googlebot I signed it!

Magic line. Perhaps this message will suffice...

ben-clayton commented:
Woo, that did the trick. Thanks!

ben-clayton commented on BUILD:
```diff
@@ -0,0 +1,49 @@
+# Copyright 2019 The Marl Authors.
```
Please can we rename this file to `BUILD.bazel`?
This collides with the default `build` output directory on case insensitive filesystems.

ben-clayton commented on examples/BUILD:
```diff
@@ -0,0 +1,33 @@
+# Copyright 2019 The Marl Authors.
```
And this file to `BUILD.bazel` for consistency's sake.

ben-clayton commented:
Perfect, thank you!
